packages/cli/tests/unit/queue.spec.ts

```diff
@@ -0,0 +1,14 @@
+import assert from 'assert';
+import { queue } from 'async';
+
+describe('async', () => {
+  it('hangs if you drain an idle queue', async () => {
+    const q = queue(async () => {}, 1);
+    expect(q.idle()).toBe(true);
+    let drained = false;
+    q.drain().then(() => (drained = true));
+    setTimeout(() => {
+      assert(!drained, 'drain should not have resolved yet');
+    }, 50);
```

Comment:
This timeout will fire after the test case returns. I suggest instead `await`ing a `setTimeout` from `timers/promises` and then asserting after that.